pkg/controller/external_nofork.go

```diff
@@ -513,3 +587,59 @@ func (n *noForkExternal) fromInstanceStateToJSONMap(newState *tf.InstanceState)
 	}
 	return stateValueMap, nil
 }
+
+// GetTerraformIgnoreChanges returns a sorted Terraform `ignore_changes`
+// lifecycle meta-argument expression by looking for differences between
+// the `initProvider` and `forProvider` maps. The ignored fields are the ones
+// that are present in initProvider, but not in forProvider.
+// TODO: This method is copy-pasted from `pkg/resource/ignored.go` and adapted.
+// Consider merging this implementation with the original one.
+func GetTerraformIgnoreChanges(forProvider, initProvider map[string]any) []string {
+	ignored := getIgnoredFieldsMap("%s", forProvider, initProvider)
+	return ignored
+}
+
+// TODO: This method is copy-pasted from `pkg/resource/ignored.go` and adapted.
+// Consider merging this implementation with the original one.
+func getIgnoredFieldsMap(format string, forProvider, initProvider map[string]any) []string {
+	ignored := []string{}
+
+	for k := range initProvider {
+		if _, ok := forProvider[k]; !ok {
+			ignored = append(ignored, fmt.Sprintf(format, k))
+		} else {
+			// both are the same type so we dont need to check for forProvider type
+			if _, ok = initProvider[k].(map[string]any); ok {
+				ignored = append(ignored, getIgnoredFieldsMap(fmt.Sprintf(format, k)+".%v", forProvider[k].(map[string]any), initProvider[k].(map[string]any))...)
+			}
+			// if its an array, we need to check if its an array of maps or not
+			if _, ok = initProvider[k].([]any); ok {
+				ignored = append(ignored, getIgnoredFieldsArray(fmt.Sprintf(format, k), forProvider[k].([]any), initProvider[k].([]any))...)
+			}
+
+		}
+	}
+	return ignored
+}
+
+// TODO: This method is copy-pasted from `pkg/resource/ignored.go` and adapted.
+// Consider merging this implementation with the original one.
+func getIgnoredFieldsArray(format string, forProvider, initProvider []any) []string {
+	ignored := []string{}
+	for i := range initProvider {
+		// Construct the full field path with array index and prefix.
+		fieldPath := fmt.Sprintf("%s[%d]", format, i)
```

Comment:
I think we should be using the `Set` function specified in the Terraform resource's schema (`SchemaSetFunc`) instead of the sequential index here for elements of type `TypeSet`.

Reply:
Yes, we should. Proposed implementation works for top-level sets, though. So we are leaving applying `SchemaSetFunc` for future, when we address nested sets.